snap/info.go

```diff
@@ -252,6 +259,13 @@ func HooksDir(name string, revision Revision) string {
 	return filepath.Join(MountDir(name, revision), "meta", "hooks")
 }
 
+// ComponentHooksDir returns the directory containing the component's hooks for
+// the given component hook name. The provided snap name can be either a snap
+// name or snap instance name.
+func ComponentHooksDir(componentName, snapInstance string, snapRevision Revision) string {
+	return filepath.Join(ComponentMountDir(componentName, snapInstance, snapRevision), "meta", "hooks")
+}
```

Comment:
wondering if the two Component*Dir functions couldn't take instead a PlaceInfo for (snapInstance, snapRevision) to avoid needing to remember which is which of the two string arguments?